src/components/content/monitor/Monitor.tsx

```diff
@@ -0,0 +1,326 @@
+/*
+ * SPDX-License-Identifier: Apache-2.0
+ * SPDX-FileCopyrightText: Huawei Inc.
+ */
+
+import 'echarts/lib/chart/bar';
+import '../../../styles/monitor.css';
+import { MonitorOutlined } from '@ant-design/icons';
+import { Alert, Button, Col, Form, Input, Row, Select, Tabs } from 'antd';
+import { Tab } from 'rc-tabs/lib/interface';
+import React, { useEffect, useState } from 'react';
+import MonitorChart from './MonitorChart';
+import { Metric, MonitorService, ServiceService, ServiceVo } from '../../../xpanse-api/generated';
+import { arrayQueue, Queue } from '../../utils/Queue';
+import { fetchMonitorMetricDataTimeInterval, monitorMetricQueueSize } from '../../utils/constants';
+
+export interface MetricProps {
```

Comment:
can we move this to a separate file?
and can this be a type instead of interface?